src/main/java/com/bekzat/gym/service/map/TrainerServiceMap.java

```diff
@@ -1,23 +1,95 @@
 import org.springframework.stereotype.Service;
 
+@Slf4j
 @Service
+@AllArgsConstructor
 public class TrainerServiceMap implements TrainerService {
-    @Autowired
-    private TrainerDAO trainerDAO;
 
-    @Override
-    public Trainer findById(Long id) {
-        return trainerDAO.findById(id).orElseThrow(() -> new RuntimeException("Trainer not found"));
+    private final TrainerRepository trainerRepository;
+    private final UserRepository userRepository;
+    private final UserCredentialsService userCredentialsService;
+    private final TrainerMapper trainerMapper;
+
+    public TrainerReadDto findById(Long id, CredentialsDto credentialsDto) {
+        log.info("Finding trainer by ID: {}", id);
+        return trainerRepository.findById(id)
+                .map(trainerMapper::toDto)
+                .orElseThrow(() -> new TrainerNotFoundException(id));
     }
+    @Transactional
+    public CredentialsDto create(TrainerCreateDto trainer) {
+        Trainer trainerEntity = trainerMapper.toEntity(trainer);
+        String username = userCredentialsService.generateUsername(trainer.firstName(), trainer.lastName());
+        String password = userCredentialsService.generateRandomPassword();
+        trainerEntity.setUsername(username);
+        trainerEntity.setPassword(password);
+        trainerRepository.save(trainerEntity);
 
-    @Override
-    public Trainer save(Trainer trainer) {
-        return trainerDAO.save(trainer);
+        return new CredentialsDto(username, password);
+    }
+    @Transactional
+    public void changeTrainerPassword(Long trainerId, String newPassword, CredentialsDto credentialsDto) {
+        if (userCredentialsService.checkCredentials(credentialsDto)) {
+            throw new AuthenticationException("Invalid credentials");
+        }
+        log.info("Checking password for trainer with trainer id: {}", trainerId);
+        userCredentialsService.changePassword(trainerId, newPassword);
+    }
+    @Transactional
+    public boolean checkTrainerCredentials(String username, String password, CredentialsDto credentialsDto) {
+        if (userCredentialsService.checkCredentials(credentialsDto)) {
+            throw new AuthenticationException("Invalid credentials");
+        }
+        log.info("Checking credentials for trainer with username: {}", username);
+        return userRepository.checkCredentials(username, password);
+    }
+    @Transactional
+    public TrainerReadDto findTrainerByUsername(String username, CredentialsDto credentialsDto) {
+
+        log.info("Finding trainer by username: {}", username);
+        return trainerRepository.findByUsername(username)
+                .map(trainerMapper::toDto)
+                .orElseThrow(() -> new UserNotFoundException(username));
+    }
+    @Transactional
+    public void activateTrainer(Long id, CredentialsDto credentialsDto) {
+        if (userCredentialsService.checkCredentials(credentialsDto)) {
```

Comment:
it is idempotent method - call this method several times and result will be the same like after first time
try to create one method to change isActive value 
if value true than set it to false, when false - set it to true